UN-OCHA/unocha-site
# [UNO-745] Cleaning up

Refs: UNO-745

This a code cleanup PR that does the following:

- Update the local stack to the latest "standards"
- Update the dockerfile and settings to the latest "standards"
- Remove unused patches and clean up the composer.json file
- Remove unused custom modules and code (unocha_figures and canto portals)
- Sync config (minor changes to order of elements in config etc.)

## Tests

Note: if you were already using the `local/` for your local stack, you may want to remove it with `./local/install.sh -x -v`. If you want to preserve it then, clone the repository to somewhere else like `unocha-site-uno-745` and adjust the site URL mentioned in the steps below to something different like `unocha-uno-745.test`.

1. Checkout the branch
2. Create local stack
   1. Follow the instruction for the local proxy in https://github.com/UN-OCHA/local-reverse-proxy/ if you don't already have it set up
   2. Read the `local/README.md` file
   3. Copy the `local/shared/settings/setting.site.php.template` to `local/shared/settings/setting.site.php` and adjust the settings
   4. Copy the `local/.env.example` to `local/.env` and edit the mapbox token for example. (Optionally change the `unocha-local.test` url to something to your liking)
   5. Add the chosen local site URL in (4) to your `/etc/hosts`
   6. Run `./local/install.sh -m -c -i` to create a new empty instance of the site
   7. Download the latest DB dump and unzip it somewhere
   8. Drop the DB tables with `./local/exec.sh -u appuser site drush sql-drop -y`
   9. Import it with `./local/exec.sh -T -u appuser site drush sqlc < path/to/unocha/db/dump.sql`
   10. Install the dev dependencies and import the config with `./local/install.sh -d`
3. Check that the site still works by clicking and editing various pages (compare to prod for example)
